tests/bluetooth/host/keys/bt_keys_foreach_bond/src/main.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/addr.h>
+#include <host/keys.h>
+#include "mocks/keys_help_utils.h"
+#include "testing_common_defs.h"
+
+/* This LUT contains different combinations of ID and Address pairs */
+static const struct id_addr_pair testing_id_addr_pair_lut[] = {
+	{ BT_ADDR_ID_1, BT_ADDR_LE_1 },
+	{ BT_ADDR_ID_1, BT_ADDR_LE_2 },
+	{ BT_ADDR_ID_2, BT_ADDR_LE_1 },
+	{ BT_ADDR_ID_2, BT_ADDR_LE_2 }
+};
+
+/* This list will hold returned references while filling keys pool */
+static struct bt_keys *returned_keys_refs[CONFIG_BT_MAX_PAIRED];
+
+BUILD_ASSERT(ARRAY_SIZE(testing_id_addr_pair_lut) == CONFIG_BT_MAX_PAIRED);
+BUILD_ASSERT(ARRAY_SIZE(testing_id_addr_pair_lut) == ARRAY_SIZE(returned_keys_refs));
+
+/* Callback to be used when no calls are expected by bt_foreach_bond() */
+static void bt_foreach_bond_unreachable_cb(const struct bt_bond_info *info, void *user_data)
+{
+	zassert_unreachable("Unexpected call to '%s()' occurred", __func__);
+}
+
+/* Callback to be used when calls are expected by bt_foreach_bond() */
+static void bt_foreach_bond_expected_cb(const struct bt_bond_info *info, void *user_data)
```

Comment:
could you place the callbacks right before the test case that uses them? it's easier to read without jumping around in the editor.